TypeCobol/Compiler/Diagnostics/CrossChecker.cs

```diff
@@ -857,9 +883,18 @@ private static bool HasChildrenThatDeclareData([NotNull] DataDefinition dataDefi
             {
                 var lastChild = dataDefinition.Children[dataDefinition.ChildrenCount - 1];
 
+                if (lastChild.CodeElement == null) //index
+                {
+                    return false;
+                }
+
+                if (lastChild.CodeElement.Type == CodeElementType.DataRenamesEntry)
+                {
+                    return dataDefinition.Children.Any(c => c is DataDescription);
+                }
```

Comment:
This part contradicts the comment written at the beginning. Also i remember thaht @smedilol avoided iteration over children collection so we should try to find another way to deal with renames (if possible)

Reply:
I suggested another comment at the beginning.
I also suggested this code.
We have few renames in our company and renames are only attached to a level 01 node. So I think it's ok to iterate over children here.

We could improve this by adding a flag `HasChildrenThatDeclaraData` or `HasDataDescriptionChildren` on the node when we attached children in `ProgramClassBuilder`.

Reply:
Code not changed but a comment explains why we iterate.